### PR TITLE
lenstool: submission (new Portfile)

### PR DESCRIPTION
**Pull request by ejullo**
###### Type(s)

- [ ] bugfix
- [ ] enhancement
- [ ] security fix
- [x] submission (new Portfile)

###### Tested on

macOS 10.13.6 17G5019
Xcode 9.1 9B55 

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?

- [x] checked your Portfile with `port lint`?
- [x] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files?

### CODE REVIEW

File: science/lenstool/Portfile [hunk 1]
```diff
@@ -0,0 +1,39 @@
+# -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
+
+PortSystem          1.0
+PortGroup           compilers 1.0
+
+name                lenstool
+version             7.1.1
+categories          science
+license             GPL
```
Comment (mf2k) [inline science/lenstool/Portfile]: We normally specify the license more precisely. Please investigate which version of the GPL they are using. Normally it is something like `GPL-2` or `GPL-2`+ (2 or later).

File: science/lenstool/Portfile [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+                    MCMC model optimization, MCMC samples are delivered in \
+                    an output ASCII file. MCMC samples can be post-processed \
+                    by several additional executables provided in the package. 
+
+homepage            https://projets.lam.fr/projects/lenstool
+master_sites        https://projets.lam.fr/attachments/download/5851
+distname            ${name}-${version}
+worksrcdir          ${name}-${version}
+
+checksums           sha256 272aed5e5964b8b941c916be98f13a7f134f3a785681c5b6199ab0438114f5ac
```
Comment (mf2k) [inline science/lenstool/Portfile]: Add an `rmd160` checksum and the `size` to the `checksums`.

File: science/lenstool/Portfile [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+                    output results in the graphical interface DS9 provided by \
+                    Smithsonian Astrophysical Observatory (SAO). In case of \
+                    MCMC model optimization, MCMC samples are delivered in \
+                    an output ASCII file. MCMC samples can be post-processed \
+                    by several additional executables provided in the package. 
+
+homepage            https://projets.lam.fr/projects/lenstool
+master_sites        https://projets.lam.fr/attachments/download/5851
+distname            ${name}-${version}
+worksrcdir          ${name}-${version}
```
Comment (mf2k) [inline science/lenstool/Portfile]: It looks like you are using the default values for both `distname` and `worksrcdir `. We normally just delete the lines in that case.

File: science/lenstool/Portfile [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+
+checksums           sha256 272aed5e5964b8b941c916be98f13a7f134f3a785681c5b6199ab0438114f5ac
+
+depends_lib         port:gsl \
```
Comment (mf2k) [inline science/lenstool/Portfile]: Change `depends_lib` to `depends_lib-append` to not overwrite the dependencies added by the `compilers` portgroup.

File: science/lenstool/Portfile [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+                    port:wcslib \
+                    port:cfitsio
```
Comment (mf2k) [inline science/lenstool/Portfile]: Please sort the dependencies by name.

File: science/lenstool/Portfile [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+distname            ${name}-${version}
+worksrcdir          ${name}-${version}
+
+checksums           sha256 272aed5e5964b8b941c916be98f13a7f134f3a785681c5b6199ab0438114f5ac
+
+depends_lib         port:gsl \
+                    port:wcslib \
+                    port:cfitsio  
+
+configure.args      --with-cfitsio-prefix=${prefix} --with-wcslib-include-path=${prefix}/include/wcslib --with-wcslib-lib-path=${prefix}/lib --with-gsl-prefix=${prefix}
```
Comment (mf2k) [inline science/lenstool/Portfile]: It is easier to read these if you list one arg per line like dependencies are listed.

File: science/lenstool/Portfile [hunk 7]
```diff
@@ -0,0 +1,39 @@
+# -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
+
+PortSystem          1.0
+PortGroup           compilers 1.0
```
Comment (reneeotten) [inline science/lenstool/Portfile]: why did you include the compilers portgroup? It doesn't look like you're actually using any of it.

Comment (mf2k) [reply science/lenstool/Portfile]: Good point! You want to call the setup procedure.

File: science/lenstool/Portfile (hunk 2 above)
Comment (mf2k) [reply science/lenstool/Portfile]: The `size` still needs to be added.

File: science/lenstool/Portfile (hunk 5 above)
Comment (mf2k) [reply science/lenstool/Portfile]: And this is not done yet.

**Comment by cjones051073** [comment]
@ejullo Please squash the two commits here into a single one.

Please also do not close this PR (like last time) to do this. If you do not know how to squash commits and then (force) push the changes to the branch here, then just ask for help.

**Comment by cjones051073** [comment]
Thanks for making the requested changes. I believe they are now all done, @mf2k do you agree ?

**Comment by mf2k** [comment]
Yes, looks good. Thank you!

**Comment by ejullo** [comment]
Thank you all for your patience.

**Comment by cjones051073** [comment]
No problem. New contributors are always very welcome.

**Pull request merged**